Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -40,41 +41,38 @@ static u64 s_title_to_launch;
 struct DirectoryToCreate
 {
   const char* path;
-  u32 attributes;
-  OpenMode owner_perm;
-  OpenMode group_perm;
-  OpenMode other_perm;
+  FS::FileAttribute attribute;
+  FS::Mode owner_mode;
+  FS::Mode group_mode;
+  FS::Mode other_mode;
+  FS::Uid uid;
+  FS::Gid gid;
 };
 
 constexpr std::array<DirectoryToCreate, 9> s_directories_to_create = {{
-    {"/sys", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_NONE},
-    {"/ticket", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_NONE},
-    {"/title", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_READ},
-    {"/shared1", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_NONE},
-    {"/shared2", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW},
-    {"/tmp", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW},
-    {"/import", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_NONE},
-    {"/meta", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_RW},
-    {"/wfs", 0, OpenMode::IOS_OPEN_RW, OpenMode::IOS_OPEN_NONE, OpenMode::IOS_OPEN_NONE},
+    {"/sys", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::None, 0, 0},
+    {"/ticket", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::None, 0, 0},
+    {"/title", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::Read, 0, 0},
+    {"/shared1", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::None, 0, 0},
+    {"/shared2", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::ReadWrite, 0, 0},
+    {"/tmp", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::ReadWrite, 0, 0},
+    {"/import", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::None, 0, 0},
+    {"/meta", 0, FS::Mode::ReadWrite, FS::Mode::ReadWrite, FS::Mode::ReadWrite, 0x1000, 1},
```

Comment:
Is `1` and `0x1000` important enough to get its own `PID_xxxx` constant?

Reply:
Turned out that this constant was already being used in uid.sys handling code, so I've changed 0x1000 to a named constant. It's the first UID assigned to PPC titles.